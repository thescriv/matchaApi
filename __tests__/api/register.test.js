const { startApi, stopApi } = require('../../src/api')
const { apiClient } = require('../apiClient')

let client

describe('Register API', () => {
  beforeAll(async () => {
    await startApi(3000)

    client = new apiClient(3000)
  })

  afterAll(async () => {
    await stopApi()
  })

  describe('POST /register', () => {
    let registerPayload

    beforeEach(() => {
      registerPayload = {
        email: 'test@test.com',
        first_name: 'test',
        last_name: 'test'
      }
    })

    test('do register user', async () => {
      const { body, status } = await client.postRegister(registerPayload)

      expect({ body, status }).toMatchSnapshot()
    })
  })
})
