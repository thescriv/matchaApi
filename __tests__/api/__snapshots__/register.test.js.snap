// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register API POST /register do register user 1`] = `
Object {
  "body": Object {},
  "status": 200,
}
`;
